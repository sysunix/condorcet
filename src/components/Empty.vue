<template>
  <div
    v-if="condition"
    class="flex flex-col justify-center items-center h-full"
  >
    <div class="text-xl">
      {{ text }}
    </div>
    <div class="mt-4">
      <router-link
        :to="to"
        class="bg-blue-500 hover:bg-blue-700 text-white font-bold py-2 px-4 rounded"
        >{{ actionText }}</router-link
      >
    </div>
  </div>
</template>

<script>
export default {
  props: {
    condition: {
      type: Boolean,
      required: true
    },
    text: {
      type: String,
      required: true
    },
    actionText: {
      type: String,
      required: true
    },
    to: {
      type: String,
      required: true
    }
  }
};
</script>

<style></style>
