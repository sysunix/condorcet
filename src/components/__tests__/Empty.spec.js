import { shallowMount } from "@vue/test-utils";
import Empty from "../Empty.vue";

describe("Empty.vue", () => {
  it("should render if condition is truthy", () => {
    const wrapper = shallowMount(Empty, {
      propsData: {
        condition: true,
        to: "/polls/new",
        text: "Smaug",
        actionText: "Create Smaug"
      },
      stubs: ["router-link"]
    });
    expect(wrapper).toMatchSnapshot();
  });

  it("should not render if condition is falsy", () => {
    const wrapper = shallowMount(Empty, {
      propsData: {
        condition: false,
        to: "/polls/new",
        text: "Smaug",
        actionText: "Create Smaug"
      },
      stubs: ["router-link"]
    });
    expect(wrapper).toMatchSnapshot();
  });
});
