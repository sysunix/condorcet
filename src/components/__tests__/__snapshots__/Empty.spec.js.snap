// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Empty.vue should not render if condition is falsy 1`] = ``;

exports[`Empty.vue should render if condition is truthy 1`] = `
<div class="flex flex-col justify-center items-center h-full">
  <div class="text-xl">
    Smaug
  </div>
  <div class="mt-4">
    <router-link-stub to="/polls/new" class="bg-blue-500 hover:bg-blue-700 text-white font-bold py-2 px-4 rounded">Create Smaug</router-link-stub>
  </div>
</div>
`;
